backend/src/common/auth.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import { AuthRequest } from '../common/auth.middleware';
+import { Request, Response, NextFunction } from 'express';
+import httpStatus from 'http-status';
+import jwt from 'jsonwebtoken';
+import { IUserDetails } from 'shared-types';
+import createLogger from '../utils/logger';
+import config from '../config';
+
+const logger = createLogger('auth middleware');
+
+// Add user property to express request interface
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Express {
+    interface Request {
+      user?: IUserDetails;
+    }
+  }
+}
+export { Request as AuthRequest };
+
+const authMiddleware = (req: AuthRequest, res: Response, next: NextFunction) => {
+  const authHeader = req.headers['authorization'];
+  const token = authHeader && authHeader.split(' ')[1]; // Bearer <token>
```

Comment:
maybe extract this line to utils (we use it multiple times)
`const token = authHeader && authHeader.split(' ')[1]; // Bearer <token>
`

Reply:
Right. we can also make this a middleware on every request to `/api`.
The middleware will ensure the bearer token and set it in `req.token` for easier access

Reply:
We don't need the token itself since only the value inside the token is meaningful. I will put it in a utils file